Extractor/ServerSubscriptionManager.cs

```diff
@@ -0,0 +1,153 @@
+                        namespaceNameToId.Add(refDef.DisplayName.ToString(), nodeId.ToString());
+                    }
+                    else if (
+                        grouping.ContainsKey(parent)
+                        // Ensures that the type of node being added is a variable node class
+                        && refDef.NodeClass == NodeClass.Variable
+                        // Filters targets nodes
+                        && targetNodes.Contains(refDef.DisplayName.ToString())
+                    )
+                    {
+                        grouping.GetValueOrDefault(parent).Add(refDef);
+                    }
+                },
+                token,
+                maxDepth: 1,
+                doFilter: false,
+                ignoreVisited: false
+            );
+
+            // To be used in filtering namespaces
+            var availableNamespaces = namespaceNameToId.Keys;
+
+            // Filters by namespaces
+            var processedNamespaces = (
+                shouldFilterNamespaces
+                    ? filteredNamespaces.Intersect(availableNamespaces)
+                    : availableNamespaces
+            ).ToList();
+
+            if (shouldFilterNamespaces && processedNamespaces.Count() < filteredNamespacesCount)
+            {
+                _logger.LogInformation(
+                    "Some namespaces could not be processed as they do not exist on the server: {namespaces}",
+                    filteredNamespaces.Except(processedNamespaces)
+                );
+            }
+
+            foreach (var @namespace in processedNamespaces)
+            {
+                var nodeId = namespaceNameToId.GetValueOrDefault(@namespace);
+                var references = grouping.GetValueOrDefault(nodeId);
+
+                nodeIds.AddRange(
+                    references.Where(@ref => targetNodes.Contains(@ref.DisplayName.ToString()))
+                        .Select(@ref => (NodeId)@ref.NodeId)
+                );
+            };
+
+            if (nodeIds.Count() > 0) _logger.LogInformation("The following nodes will be subscribed to a rebrowse: {nodes}", nodeIds);
+
+            var nodes = nodeIds.Select(node => new ServerItemSubscriptionState(_uaClient, node)).ToList();
+
+            await CreateSubscriptions(nodes, token);
+        }
+
+        private async Task CreateSubscriptions(List<ServerItemSubscriptionState> nodes, CancellationToken token)
+        {
+            var sub = await _uaClient.AddSubscriptions(
+                nodes, "TriggerRebrowse",
+                async (MonitoredItem item, MonitoredItemNotificationEventArgs _) =>
+                {
+                    var readValueId = new [] { 
+                        new ReadValueId { NodeId = item.ResolvedNodeId, AttributeId = Attributes.Value }
+                    };
+                    var read = await _uaClient.ReadAttributes(new ReadValueIdCollection(readValueId), 1, token);
```

Comment:
This you don't need to do. See how this is done in `Streamer`. You can use `DequeueValues`, the values are passed along with the notification.